js/modules/k6/grpc/client.go

```diff
@@ -461,7 +461,15 @@ func (c *Client) parseInvokeParams(paramsVal goja.Value) (*invokeParams, error)
 				// TODO(rogchap): Should we manage a string slice?
 				strval, ok := kv.(string)
 				if !ok {
-					return result, fmt.Errorf("metadata %q value must be a string", hk)
+					if strings.HasSuffix(hk, "-bin") {
```

Comment:
```suggestion
                    // The spec defines that Binary-valued keys end in -bin
                    //  https://grpc.io/docs/what-is-grpc/core-concepts/#metadata
					if strings.HasSuffix(hk, "-bin") {
```


I would prefer if we mention directly the spec in a comment so for the next contributor is clear that k6 is just following the spec and we haven't applied an arbitrary decision.